portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -79,22 +95,41 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType(contentType);
     String name = getParameter(request, "name", "Anonymous");
-    String comment = getParameter(request, "comment", "");
+    String comment = getParameter(request, "text", "");
     Date createDate = new Date();
 
     if (comment == "" || comment == null) {
       throw new IOException("Comment is blank");
     }
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Entity commentEntity = new Entity("Comment");
-    commentEntity.setProperty("name", name);
-    commentEntity.setProperty("text", comment);
-    commentEntity.setProperty("date", createDate);
-
-    datastore.put(commentEntity);
-
-    response.sendRedirect("/");
+    String idTokenString = request.getParameter("id_token");
+    
+    try {
+      JacksonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+      NetHttpTransport transport = new NetHttpTransport();
+      
+      GoogleIdToken idToken = validate(idTokenString, jsonFactory, transport);
+
+      if (idToken != null) {
+        Payload payload = idToken.getPayload();
+        
+        String userId = payload.getSubject();
```

Comment:
Do you use it anywhere? If not at this moment, what would be a good future use for this field?

Reply:
Thinking about it, I don't think userId is necessary, since I already stored emails in the datastore, and those are unique anyway; will delete it in a future commit. Can't think of a situation where the userId can't be replaced with the email.